sound/soc/sof/sof-priv.h

```diff
@@ -308,6 +308,11 @@ struct snd_sof_pcm_stream {
 	struct snd_pcm_substream *substream;
 	struct work_struct period_elapsed_work;
 	bool d0i3_compatible; /* DSP can be in D0I3 when this pcm is opened */
+	/*
```

Comment:
the commit message makes no sense, you are again confusing stream and device states.

ASoC: SOF: PCM: add a suspend_ignored flag

Some PCM streams may need to remain in RUNNING state after a power state transition, e.g. to perform always-on tasks when the device is in D0i3 substate. Add a flag to avoid suspending those PCM streams.

I have no idea what the second paragraph was supposed to mean.

Reply:
> the commit message makes no sense, you are again confusing stream and device states.
> 
> ASoC: SOF: PCM: add a suspend_ignored flag
> 
> Some PCM streams may need to remain in RUNNING state after a power state transition, e.g. to perform always-on tasks when the device is in D0i3 substate. Add a flag to avoid suspending those PCM streams.

Um, see below.

> 
> I have no idea what the second paragraph was supposed to mean.

It is actually from Ranjani, I am agree with her that it is not accurate to say that the PCM stream remains in RUNNING state, from the view of ALSA, those PCMs are actually suspended, what keep running is only the pipelines inside the FW, all of these are actually transparent to ALSA.

Reply:
@plbossart as far as ALSA is concerned the stream is in the SUSPENDED state. But internally SOF prevents the pipeline from being stopped. 

I would word the message like this:
"Add a suspend_ignored flag to snd_sof_pcm_stream that will be used to decide if the corresponding FW pipeline should be kept active to perform always on tasks when the system is entering the S0ix state"

Reply:
@ranj063 I may be a bit confused here. Does the transition to SUSPENDED happen prior to calling sof_pcm_trigger, or after? the core has all these pre-do-post ops and I am not sure the state is actually SUSPENDED while you call sof_pcm_trigger.